neurom/core/morphology.py

```diff
@@ -200,6 +200,10 @@ def __repr__(self):
         return (f'Section(id={self.id}, type={self.type}, n_points={len(self.points)})'
                 f'<parent: Section(id={parent_id}), nchildren: {len(self.children)}>')
 
+    def iter_segments(self):
+        """Iterator to the segments of the section."""
+        return iter_segments(self)
+
```

Comment:
Does the method add anything other than convenience? On top of that is a limited version of the function `iter_segments` as it uses the defaults. Why not just use the function instead?

Reply:
This one is just for convenience and consistency (since for other objects we can have methods to iterate over the lower scale objects, e.g. `iter_sections()` in the `Neurite` class).

Reply:
I wish I could remove all the iter methods from the objects. :D 

ok fair enough

Reply:
Ahah yeah, that could be another solution :)
I don't really care, I just noticed it was not consistent so I added it but we could indeed remove all the `iter_*` methods.